yeat/Snakefile

```diff
@@ -79,15 +80,27 @@ rule downsample:
         read1="seq/fastp/{sample}.R1.fq.gz",
         read2="seq/fastp/{sample}.R2.fq.gz",
         mash_report="seq/mash/{sample}.report.tsv"
+    params:
+        downsample=config["downsample"]
     run:
+        if params.downsample == -1:
+            p = Path("seq/downsample")
+            p.mkdir(parents=True, exist_ok=True)
+            copyfile(input.read1, output.sub1)
+            copyfile(input.read2, output.sub2)
+            return
         result = input.mash_report
         df = pd.read_csv(result, sep="\t")
         genome_size = df["Length"].iloc[0]
-        coverage = 150
-        read_length = 250
-        down = (genome_size * 150) // (2 * read_length)
-        print(f"genome size: {genome_size}")
-        print(f"down: {down}")
+        print(f"genome size: {genome_size}")    
+        if params.downsample == 0:
+            coverage = 150
+            read_length = 250
+            down = (genome_size * coverage) // (2 * read_length)
```

Comment:
You may want to make 150 a parameter that the user can configure with the CLI.

Reply:
Also, read length for autosampling is...tricky. If they're untrimmed 250 bp reads that's great, but if they're not 250 bp read or if they're trimmed, you'd need to determine the distribution of the read lengths and use the mean or median of that distribution to determine the *approximate* number of reads needed for the desired coverage. A little more involved, and something that might benefit from a dedicated branch/thread.

Reply:
This has always bothered me a bit, too. We don't need a ton of precision, so it's OK to just land in the right ballpark (125x tends to be good enough if I asked for 150x). That said, we just had a run where the fragment size was so small that a 2x250 run was essentially 2x150 after trimming, and that's not all that uncommon. In the past, we've handled it by just asking for more coverage when we knew the fragment sizes were small, but that's not a terribly satisfying solution.